azuread/fedauth_adal.go

```diff
@@ -0,0 +1,136 @@
+package azuread
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+type azureFedAuthConfigurer struct{}
+
+func init() {
+	mssql.SetFederatedAuthenticationConfigurer(&azureFedAuthConfigurer{})
+}
+
+// When the security token library is used, the token is obtained without input
+// from the server, so the AD endpoint and Azure SQL resource URI are provided
+// from the constants below.
+const (
+	// activeDirectoryEndpoint is the security token service URL to use when
+	// the server does not provide the URL.
+	activeDirectoryEndpoint = "https://login.microsoftonline.com/"
+
+	// azureSQLResource is the AD resource to use when the server does not
+	// provide the resource.
+	azureSQLResource = "https://database.windows.net/"
```

Comment:
this needs to be configurable.
not all azure cloud have the same database resource endpoint.
I would say it's a good default, but it would be nice to be able to modify it via an environment variable, for example

Reply:
use this func instead to get the correct endpoint : 

https://github.com/Azure/go-autorest/blob/1d7168cf650ab7fdc4e05297b7b0250be08d0a8a/autorest/azure/environments.go#L227

`SQLDatabaseDNSSuffix` has the correct url on the environment struct

Reply:
or better yet, infer it from the connection string?